ci-operator/step-registry/cucushift/installer/check/aws/byo-kms-key/cucushift-installer-check-aws-byo-kms-key-commands.sh

```diff
@@ -0,0 +1,156 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+if [ -f "${SHARED_DIR}/kubeconfig" ] ; then
+  export KUBECONFIG=${SHARED_DIR}/kubeconfig
+else
+  echo "No KUBECONFIG found, exit now"
+  exit 1
+fi
+
+if test -f "${SHARED_DIR}/proxy-conf.sh"
+then
+    # shellcheck disable=SC1090
+    source "${SHARED_DIR}/proxy-conf.sh"
+fi
+
+
+REGION="${LEASED_RESOURCE}"
+INFRA_ID=$(jq -r '.infraID' ${SHARED_DIR}/metadata.json)
+CONFIG=${SHARED_DIR}/install-config.yaml
+export AWS_SHARED_CREDENTIALS_FILE="${CLUSTER_PROFILE_DIR}/.awscred"
+
+
+function is_empty()
+{
+    local v="$1"
+    if [[ "$v" == "" ]] || [[ "$v" == "null" ]]; then
+        return 0
+    fi
+    return 1
+}
+
+ret=0
+
+echo "-------------------------------------------------------------"
+echo "Default EC2 KMS key on $REGION"
+echo "-------------------------------------------------------------"
+
+# default KMS key used by EC2
+default_kms_key_id=$(aws --region $REGION ec2 get-ebs-default-kms-key-id | jq -r '.KmsKeyId')
+default_kms_key_arn=$(aws --region $REGION kms describe-key --key-id ${default_kms_key_id} | jq -r '.KeyMetadata.Arn')
+
+function show_key()
+{
+  local k=${1:-}
+
+  if [[ "${k}" == "${default_kms_key_arn}" ]]; then
+    k="$k (default key)"
+  fi
+  echo $k | awk -F'/' '{print $2}'
```

Comment:
Here has some problem when `ENABLE_AWS_KMS_KEY_DEFAULT_MACHINE` enabled, e.g: https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/58439/rehearse-58439-periodic-ci-openshift-openshift-tests-private-release-4.18-multi-nightly-aws-ipi-byo-kms-etcd-encryption-arm-f2/1860881446256775168

Reply:
that's caused by incorrect `echo $key_id > "${arn_output}"` https://github.com/openshift/release/pull/58439#discussion_r1857674133